Controllers/user.js

```diff
@@ -0,0 +1,76 @@
+const Address = require('../Database/address');
+const Cart = require('../Database/cart');
+const User = require('../Database/user');
+const Wishlist = require('../Database/wishlist');
+var jwt = require('jsonwebtoken');
+
+exports.signupHandler = async (req, res) => {
+	const user = new User(req.body);
+	try {
+		const token = jwt.sign({ userId: user._id }, 'secret');
+		const savedUser = await user.save();
+		const userCart = new Cart({
+			_id: savedUser._id,
+		});
+		const savedCart = await userCart.save();
+		const userWishlist = new Wishlist({
+			_id: savedUser._id,
+		});
+		const savedWishlist = await userWishlist.save();
+		const userAddress = new Address({
+			_id: savedUser._id,
+		});
+		const savedUserAddress = await userAddress.save();
+		await Cart.findOne({ _id: savedCart._id })
+			.populate('_id')
+			.exec(function (err, usercart) {
+				console.log('The author is %s', usercart);
+			});
+
+		await Wishlist.findOne({ _id: savedWishlist._id })
+			.populate('_id')
+			.exec(function (err, userWishlist) {
+				console.log('The author is %s', userWishlist);
+			});
+
+		await Address.findOne({ _id: savedUserAddress._id })
+			.populate('_id')
+			.exec(function (err, userAddress) {
+				console.log('The author is %s', userAddress);
+			});
+
+		res.json({
+			token,
+			user: savedUser,
+			userCart: userCart,
+			userWishlist: userWishlist,
+			userAddress: userAddress,
+		});
+	} catch (error) {
+		res.json({ success: false, error: error.message });
+	}
+};
+
+exports.loginHandler = async (req, res) => {
```

Comment:
Using handler in the name of the function is not the best practice, you can use loginUser something like this which clearly tells the work/action of the function

Reply:
changed